tmt/steps/provision/__init__.py

```diff
@@ -1177,6 +1177,8 @@ def _ssh_options(self) -> Command:
             # received from the server for a long time (#868).
             '-oServerAliveInterval=60',
             '-oServerAliveCountMax=5',
+            # Avoid noisy ssh client messages spoiling test output (#2429).
+            '-o LogLevel=QUIET'
```

Comment:
Would it make sense to not add this line when debugging is enabled? E.g. `if self.debug_level > 0`. I agree that under normal conditions, this is probably spoiling the test output, and I would like to find a way to disable quiet logging just for the test, leaving it for other SSH operations, but it looks way too complicated. Keeping it enabled when debugging is enabled with `-d` or `TMT_DEBUG` would be a nice compromise.

Reply:
But this spoils output in general?

See:

https://artifacts.dev.testing-farm.io/23242cbe-7782-48fa-ae2b-b17167a8b8f7/

Which runs with `--verbose` only.
What would be the benefit of enabling this only during debug?

Reply:
I don't follow. Is that pipeline a good example, or a bad example?

> What would be the benefit of enabling this only during debug?

Your patch silences it everywhere. My proposal was to let me enable it if needed for debugging, e.g. when dealing with issues like https://github.com/teemtee/tmt/issues/2367#issuecomment-1746776600. I would very much like SSH to not be silent about troubles in cases like that, without the need to edit the tmt sources. I guess the benefit would be in you getting nicer test output and me having a way to unsilence SSH when needed.

Reply:
I see, right, I misread your comment :)

Reply:
@happz any pointers how to make now the error more visible? I was looking at the code, and I want to discuss before. So with this patch:

````
plan failed

The exception was caused by 1 earlier exceptions

Cause number 1:

    prepare step failed

    The exception was caused by 1 earlier exceptions

    Cause number 1:

        Command 'yum install -y rsync' returned 255.

        stderr (1 lines)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Unable to negotiate with 37.205.9.132 port 6929: no matching host key type found. Their offer: ssh-rsa,ssh-dss
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
````

Will become:

```
plan failed

The exception was caused by 1 earlier exceptions

Cause number 1:

    prepare step failed

    The exception was caused by 1 earlier exceptions

    Cause number 1:

        Command 'yum install -y rsync' returned 255.
```

I wanted to print something like:

```
plan failed

The exception was caused by 1 earlier exceptions

Cause number 1:

    prepare step failed

    The exception was caused by 1 earlier exceptions

    Cause number 1:

        Command 'yum install -y rsync' returned 255.

        stderr (1 lines)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        SSH connection failed, please rerun in debug mode to see the error.
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

So that would need some changes to instruct the execution to force some message?

I wonder what could be a better solution, I was thinking about adding `-oLogLevel=QUIET` only for test execution, not for these preparation commands. I expect in general issues like these would be encountered sooner, so have the ssh client hidden just during test execution would provide a reasonable way forward without complicating too much the code, wdyt?

Reply:
Yeah, setting the log level to `QUIET` only for the duration of a test would be nice. `Guest.execute()` has a `test_session` parameter, I suppose `GuestSsh` could add this `-o` into SSH options when it's set to `True`. Does it make sense?

Reply:
ack, will try, sounds like a plan

Reply:
@happz ok, it is now done only during test session